### PR TITLE
Replace obsolete posted sort field by created in mod\photos

### LINKED ISSUE #11635: Unknown column 'posted' in order clause
From the log of Friendica 2022.06:

```
[ERROR]: DB Error {
    "code": 1054,
    "error": "Unknown column 'posted' in 'order clause'",
    "callstack": "Photo::selectToArray, photos_content, LegacyModule::runModuleFunction, LegacyModule::content, BaseModule::run, App::runFrontend",
    "params": "SELECT `resource-id` FROM `photo` WHERE (`album` = 'Profile Photos' AND `uid` = 1 AND `scale` = 0 AND (allow_cid = '' AND allow_gid = '' AND deny_cid = '' AND deny_gid = '') ) ORDER BY `posted`"
} - {
    "file": "Database.php",
    "line": 679,
    "function": "p",
    "uid": "7d651c",
    "process_id": 62982
}
```

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #11635

### CODE REVIEW

File: src/Content/Text/BBCode.php [hunk 1]
```diff
@@ -995,29 +995,46 @@ private static function interpolateSavedImagesIntoItemBody($uriid, $body, array
...
+	 *               - shared : Text inside the share tags
+	 *               - author : (Optional) Display name of the shared author
+	 *               - profile: (Optional) Profile page URL of the shared author
+	 *               - avatar : (Optional) Profile picture URL of the shared author
+	 *               - link   : (Optional) Canonical URL of the shared post
+	 *               - posted : (Optional) Date the shared post was initially posted ("Y-m-d H:i:s" in GMT)
+	 *               - guid   : (Optional) Shared post GUID if any
 	 */
-	public static function fetchShareAttributes($text)
+	public static function fetchShareAttributes(string $text): array
```
Comment (nupplaphil) [inline src/Content/Text/BBCode.php:1009]: This seems like a perfect Situation for unit tests, doesn't it? ;-)

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Content/Text/BBCode.php (hunk 1 above)
Comment (MrPetovan) [reply src/Content/Text/BBCode.php:1009]: Yes!

Comment (Quix0r) [reply src/Content/Text/BBCode.php:1009]: And I just missed this one. Could catch!

Comment (Quix0r) [reply src/Content/Text/BBCode.php:1009]: If you can, please put a method description up here.

**Review by nupplaphil — APPROVED** [review]

**Pull request merged**